File: CS/Reorder/DataHelper.cs
using System;
using System.Collections.Generic;

namespace Reorder {
    public class SampleObject {
        public string Text { get; set; }
        public int GroupId { get; set; }
        public int IndexInGroup { get; set; }
    }

    public class DataHelper {
        public static List<SampleObject> GetData() {
            var data = new List<SampleObject>();
            var rnd = new Random();
            for(int n = 0; n < 32; n++) {
                var record = new SampleObject();
                record.GroupId = rnd.Next(0, 5);
                record.IndexInGroup = n;
                record.Text = "Item " + n;
                data.Add(record);
            }
            return data;
        }
    }
}

File: VB/Reorder/DataHelper.vb
Imports System
Imports System.Collections.Generic

Namespace Reorder

    Public Class SampleObject

        Public Property Text As String

        Public Property GroupId As Integer

        Public Property IndexInGroup As Integer
    End Class

    Public Class DataHelper

        Public Shared Function GetData() As List(Of SampleObject)
            Dim data = New List(Of SampleObject)()
            Dim rnd = New Random()
            For n As Integer = 0 To 32 - 1
                Dim record = New SampleObject()
                record.GroupId = rnd.Next(0, 5)
                record.IndexInGroup = n
                record.Text = "Item " & n
                data.Add(record)
            Next

            Return data
        End Function
    End Class
End Namespace
